### PR TITLE
feat: reduce noisy proxy error log

### PR DESCRIPTION
**Pull request by zoumo**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind feature

**What this PR does / why we need it**:
reduce useless proxy error log

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cmd/kube-gateway/app/proxy.go [hunk 1]
```diff
@@ -132,3 +140,16 @@ func buildProxyHandlerChainFunc(clusterManager clusters.Manager, enableAccessLog
...
 	return nativeopenapi.GetOpenAPIDefinitions(ref)
 }
+
+// proxyHTTPErrorLogWriter serves as a bridge between the standard log package and the klog package.
+// It also filter out some noisy http error log
+type proxyHTTPErrorLogWriter struct{}
+
+// Write implements the io.Writer interface.
+func (writer proxyHTTPErrorLogWriter) Write(data []byte) (n int, err error) {
+	if bytes.HasPrefix(data, []byte("http: TLS handshake error from")) {
```
Comment (xuchen-xiaoying) [inline cmd/kube-gateway/app/proxy.go:150]: will it have effect if data is cut off into segments?

File: hack/debug.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
```
Comment (xuchen-xiaoying) [inline hack/debug.sh:1]: a little confused about the use of this script

File: pkg/gateway/proxy/dispatcher/upgradeaware.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	newReq.Header = utilnet.CloneHeader(req.Header)
+	if !h.UseRequestLocation {
+		newReq.URL = &loc
+	}
+
+	proxy := httputil.NewSingleHostReverseProxy(&url.URL{Scheme: h.Location.Scheme, Host: h.Location.Host})
+	proxy.Transport = h.Transport
+	proxy.FlushInterval = h.FlushInterval
+	proxy.ErrorLog = log.New(noSuppressPanicError{}, "", log.LstdFlags)
+	if h.Responder != nil {
```
Comment (xuchen-xiaoying) [inline pkg/gateway/proxy/dispatcher/upgradeaware.go:98]: h.Responder.Error != nil ?

File: cmd/kube-gateway/app/proxy.go (hunk 1 above)
Comment (zoumo) [reply cmd/kube-gateway/app/proxy.go:150]: this http error mesage is written all at once from http pakcage, will not be cut into segments

File: hack/debug.sh (hunk 2 above)
Comment (zoumo) [reply hack/debug.sh:1]: the script starts a kube-gateway process locally for debugging

File: pkg/gateway/proxy/dispatcher/upgradeaware.go (hunk 3 above)
Comment (zoumo) [reply pkg/gateway/proxy/dispatcher/upgradeaware.go:98]: h.Responder is an Interface, h.Responder.Error will not be nil if h.Responder != nil

File: cmd/kube-gateway/app/proxy.go (hunk 1 above)
Comment (xuchen-xiaoying) [reply cmd/kube-gateway/app/proxy.go:150]: get it

File: hack/debug.sh (hunk 2 above)
Comment (xuchen-xiaoying) [reply hack/debug.sh:1]: should we add illustration or comment in readme or somewhere else?

File: pkg/gateway/proxy/dispatcher/upgradeaware.go (hunk 3 above)
Comment (xuchen-xiaoying) [reply pkg/gateway/proxy/dispatcher/upgradeaware.go:98]: oh I thought it was a pointer field. but line 56 in this file call it directly without nil check.

File: hack/debug.sh (hunk 2 above)
Comment (zoumo) [reply hack/debug.sh:1]: ok

Comment (zoumo) [reply hack/debug.sh:1]: I will revert debug script commit and push a new PR later

File: pkg/gateway/proxy/dispatcher/upgradeaware.go (hunk 3 above)
Comment (zoumo) [reply pkg/gateway/proxy/dispatcher/upgradeaware.go:98]: line 56 is copied from k8s.io/apimachinery
Responder will never be nil in this case

**Review by xuchen-xiaoying — APPROVED** [review]
LGTM

**Pull request merged**